### PR TITLE
src/util: Improve min/max

### PR DESCRIPTION
**Pull request by NickeZ**
This stops double evaluation of arguments to min/max. With previous macro, `MIN(i++,16)` would have increased `i` twice.

### CODE REVIEW

**Comment by benma** [comment]
@NickeZ A thought: wouldn't it be easier to just turn those into functions instead of macros? Maybe an `inline static` full function in the header so it is inlined like the macro is.

**Comment by NickeZ** [comment]
> @NickeZ A thought: wouldn't it be easier to just turn those into functions instead of macros? Maybe an `inline static` full function in the header so it is inlined like the macro is.

I thought about that as well, but since C doesn't have generics you cannot have min/max functions that work with different types :/

File: test/unit-test/test_util.c [hunk 1]
```diff
@@ -0,0 +1,61 @@
+#include "util.h"
+
+// Copyright 2019 Shift Cryptosecurity AG
```
Comment (benma) [inline test/unit-test/test_util.c]: license is usually the very first thing in the file

File: test/unit-test/test_util.c [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+#include <stddef.h>
+#include <cmocka.h>
+
+#include "util.h"
+
+static void test_minmax(void** state)
+{
+    (void)state;
+
+    assert_true(1);
```
Comment (benma) [inline test/unit-test/test_util.c]: ??

File: test/unit-test/test_util.c [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    assert_int_equal(res, 10);
+
+    res = MAX(10, 5);
+    assert_int_equal(res, 10);
+
+    res = MIN(5, 10);
+    assert_int_not_equal(res, 10);
+
+    res = MIN(10, 5);
+    assert_int_not_equal(res, 10);
```
Comment (benma) [inline test/unit-test/test_util.c]: redudant with line 37? this is basically testing that 10 !=5 ;)

same with the following two

**Pull request merged**